docs/enhancements/ai-on-zero-worker-hypershift.md

```diff
@@ -0,0 +1,121 @@
+
+Open Issue: How and where is that secret provisioned and become known to us (AI/ACM integration issue)
+
+#### Separating L0 and L1 resources
+The resources to watch will be created by a factory and divided into L0 and L1 resources. 
+
+The infrastructure controller then uses the appropriate clients and the inventory of resources to provision and update the necessary configuration according to the scheme above.
+
+#### RBAC
+On L1 we need the following permissions:
+- To support admission webhook configuration, add a ClusterRole to access validating/mutating webhook configurations, subjectaccessreviews
+- To support the service's controllers, add a ClusterRole to access secrets, routes, nodes, pods, endpoints, services, etc.
+
+This is a mirroring of the RBAC state on L0 in non-hypershift setup.
+
+On L0 we need a few permission that previously were defined by the OLM to allow the service to operate:
+- To support the service's leader election mechanism a Role to access leases and configmaps
+
+Open Issue: Can the leader-election-related roles be defined only on L1?
+
+### Webhooks
+Reference: [Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
+Admission webhooks are HTTP callbacks that receive admission requests and do something with them. You can define two types of admission webhooks, validating admission webhook and mutating admission webhook. To register admission webhooks, create MutatingWebhookConfiguration or ValidatingWebhookConfiguration API objects.
+These configuration objects need to be registered with L1 api-server.
+
+The admission logic is served by an admission webhook server which serves as an extension to the api server. The pod that host the webhook server is deployed on the management cluster (L0).
+
+In our current assisted-service deployment the communication between the api-server and the webhook server is done by a service since both runs on the same cluster. In the hypershift case, it should be done via konnectivity agent and client.
+
+To accomplish that we need to:
+1. On L0: Deploy a service that exposes the webhook admission server
+2. On L1: Define a headless service with endpoint that points to the L0 service's ClusterIP
+3. Specify the L1 service as the connection point to the webhook on the webhook configuration
+
+Konnectivity server on the hostedCluster (hypershift) will forward the communication to the agent on the
+management cluster and from there using the L0 service it will be forwarded to the admision pod.
+
+RBAC definition for the webhook need now be defined on both L0 and L1 (where in the central implementation they were defined only on L0)
+
+## Implementation Details
+### HypershiftAgentClusterConfig CRD
+```
+type HypershiftAgentServiceConfigSpec struct {
+	AgentServiceConfigSpec <-- re-using and keeping track of the configuration details of AgentClusterConfig
```

Comment:
We probably just want to embed this resource, rather than make it a field and sub-object.

Reply:
LGTM. My recollection is that it looked different when I reviewed it, but maybe I just misinterpreted.